src/ucp/tag/eager_rcv.c

```diff
@@ -105,9 +105,8 @@ ucp_eager_tagged_handler(void *arg, void *data, size_t length, unsigned am_flags
                                                   recv_len, 0, 1);
             ucp_request_complete_tag_recv(req, status);
         } else {
-            eagerf_hdr                = data;
-            req->recv.tag.info.length =
-            req->recv.tag.remaining   = eagerf_hdr->total_len;
+            eagerf_hdr = data;
+            ucp_tag_recv_process_eager_first_hdr(data, req);
```

Comment:
it is dangerous to initialize `req->recv.tag.non_contig_buf` here, because it is expected flow (`req->recv.tag.rdesc` may be already initialized)

Reply:
yes, better to not do it